### PR TITLE
Extended family search to filter on geography & status.

### PR DESCRIPTION
**Pull request by katybaulch**
# Description

Added family status and geography filtering to search endpoint.

## Type of change

Please select the option(s) below that are most relevant:

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change

## How Has This Been Tested?

Unit & integration.

## Reviewer Checklist

- [ ] The PR represents a single feature (small driveby fixes are also ok)
- [ ] The PR includes tests that are sufficient for the level of risk
- [ ] The code is sufficiently commented, particularly in hard-to-understand areas
- [ ] Any required documentation updates have been made
- [ ] Any TODOs added are captured in future tickets
- [ ] No FIXMEs remain

### CODE REVIEW

File: app/clients/db/models/law_policy/family.py [hunk 1]
```diff
@@ -78,6 +78,47 @@ def family_status(self) -> FamilyStatus:
...
+            .where(
+                sa.and_(
+                    FamilyDocument.family_import_id == cls.import_id,
+                    FamilyDocument.document_status == DocumentStatus.CREATED,
+                )
+            )
+            .as_scalar()
+        )
+
+        # DO NOT USE 'is None'! # type: ignore
```
Comment (katybaulch) [inline app/clients/db/models/law_policy/family.py:105]: Remove the type ignore comment here.

File: app/clients/db/models/law_policy/family.py [hunk 2]
```diff
@@ -65,7 +65,7 @@ class Family(Base):
     )
 
     @hybrid_property
-    def family_status(self) -> FamilyStatus:
+    def family_status(self) -> FamilyStatus:  # type: ignore
```
Comment (diversemix) [inline app/clients/db/models/law_policy/family.py:68]: Remember these files are copied from the backend - so need making any chanages in that repo

File: app/repository/family.py [hunk 3]
```diff
@@ -157,18 +156,42 @@ def get(db: Session, import_id: str) -> Optional[FamilyReadDTO]:
     return _family_to_dto(db, fam_geo_meta)
 
 
-def search(db: Session, search_term: str) -> list[FamilyReadDTO]:
+def search(db: Session, query_params: dict[str, str]) -> list[FamilyReadDTO]:
```
Comment (diversemix) [inline app/repository/family.py:159]: Nice - as you say would be good to more strongly type this

File: app/clients/db/models/law_policy/family.py (hunk 2 above)
Comment (katybaulch) [reply app/clients/db/models/law_policy/family.py:68]: Noted - will do this now!

Comment (katybaulch) [reply app/clients/db/models/law_policy/family.py:68]: https://github.com/climatepolicyradar/navigator-backend/pull/180

**Review by PatFawbertMills — APPROVED** [review]
Looks good

**Pull request merged**